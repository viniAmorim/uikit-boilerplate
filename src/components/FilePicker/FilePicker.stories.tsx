import React from 'react';

import { Story, Meta } from '@storybook/react';

import FilePicker, { FilePickerProps } from './FilePicker';

export default {
  title: 'Atoms/FilePicker',
  component: FilePicker,
} as Meta;

const Template: Story<FilePickerProps> = (args) => (
  <div style={{ width: 'fit-content' }}>
    <FilePicker {...args} />
  </div>
);

export const Default = Template.bind({});
Default.args = {
  children: 'Selecione um arquivo',
};
