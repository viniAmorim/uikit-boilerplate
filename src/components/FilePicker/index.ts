export { default } from './FilePicker';
export * from './FilePicker';
