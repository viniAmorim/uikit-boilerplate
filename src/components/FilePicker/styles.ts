import styled from 'styled-components';

export const FilePicker = styled.label`
  border: 1px solid ${({ theme }) => theme.colors.pink500};
  border-radius: 4px;
  padding: 10px 15px;
  cursor: pointer;
  display: flex;
  align-items: center;

  background-color: ${({ theme }) => theme.colors.white100};
  color: ${({ theme }) => theme.colors.pink500};
  font-family: 'Montserrat';
  font-weight: 700;
  font-size: 12px;
  transition: background-color 0.3s;

  &:hover {
    background-color: ${({ theme }) => theme.colors.pink500};
    color: ${({ theme }) => theme.colors.white100};
  }

  input[type='file'] {
    display: none;
  }
`;
