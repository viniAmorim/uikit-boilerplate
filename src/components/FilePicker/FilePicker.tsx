import React, { InputHTMLAttributes, ReactNode } from 'react';

import * as Styled from './styles';

export interface FilePickerProps extends InputHTMLAttributes<HTMLInputElement> {
  onFileSelect: (file: File | null) => void;
  accept?: string | undefined;
  children: ReactNode;
}

const FilePicker = ({
  onFileSelect,
  accept,
  children,
  ...rest
}: FilePickerProps) => {
  const handleFileChange = (e: React.ChangeEvent<HTMLInputElement>) => {
    const file = e.target.files?.[0] || null;
    if (file) {
      onFileSelect(file);
    }
  };

  return (
    <Styled.FilePicker data-testid="filepicker-container">
      <input
        type="file"
        accept={accept}
        onChange={handleFileChange}
        {...rest}
      />
      {children}
    </Styled.FilePicker>
  );
};

export default FilePicker;
