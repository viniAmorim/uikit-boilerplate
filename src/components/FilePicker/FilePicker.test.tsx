import React from 'react';

import { fireEvent, render } from '~/test/utils';

import FilePicker from '.';

describe('FilePicker', () => {
  it('should render correctly', () => {
    const { getByTestId } = render(
      <FilePicker onFileSelect={jest.fn()} data-testid="filepicker">
        Filepicker
      </FilePicker>,
    );

    const filepickerContainer = getByTestId('filepicker-container');

    expect(filepickerContainer).toBeInTheDocument();
    expect(filepickerContainer).toHaveTextContent(/^Filepicker$/);
    expect(getByTestId('filepicker')).toHaveAttribute('type', 'file');

    expect(filepickerContainer).toMatchSnapshot();
  });

  it('should call onFileSelect when a file is selected', () => {
    const onFileSelectMock = jest.fn();

    const { getByLabelText } = render(
      <FilePicker onFileSelect={onFileSelectMock} accept=".jpg, .png">
        Selecione um arquivo
      </FilePicker>,
    );

    const input = getByLabelText('Selecione um arquivo');

    const file = new File(['boilerplate'], 'teste.jpg', {
      type: 'image/jpeg',
    });

    fireEvent.change(input, { target: { files: [file] } });

    expect(onFileSelectMock).toHaveBeenCalledWith(file);
  });
});
