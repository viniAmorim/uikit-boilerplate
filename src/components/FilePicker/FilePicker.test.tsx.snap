// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FilePicker should render correctly 1`] = `
<label
  class="sc-beqWaB eTnXSi"
  data-testid="filepicker-container"
>
  <input
    data-testid="filepicker"
    type="file"
  />
  Filepicker
</label>
`;
